src/adafruit_blinka/microcontroller/raspi_23/spi.py

```diff
@@ -82,8 +85,8 @@ def write_readinto(self, buffer_out, buffer_in, out_start=0, out_end=None, in_st
             self._spi.max_speed_hz = self.baudrate
             self._spi.mode = self.mode
             self._spi.bits_per_word = self.bits
-            data = self._spi.xfer(list(buffer_out))
-            for i in range(len(buffer_in)): # 'readinto' the given buffer
+            data = self._spi.xfer(list(buffer_out[out_start:out_end]))
+            for i in range(len(buffer_in[in_start:in_end])): # 'readinto' the given buffer
```

Comment:
i dont think this section does what you think it does - please print/debug to make sure the data is going into the buffer_in slice!